src/Common/src/System/Drawing/KnownColorTable.cs

```diff
@@ -59,9 +55,25 @@ private static void InitColorTable()
         {
             int[] values = new int[(unchecked((int)KnownColor.MenuHighlight)) + 1];
 
-#if FEATURE_SYSTEM_EVENTS
-            SystemEvents.UserPreferenceChanging += new UserPreferenceChangingEventHandler(OnUserPreferenceChanging);
-#endif
+            // Previously we only cared about this support for SystemColors which were in Drawing.Common, so we ifdef'ed that linkage.
+            // Now that we've pushed the SystemColors to System.Drawing.Primitives, we need to remove the ifdef.
+            // To avoid pulling Microsoft.Win32.SystemEvents into the shared framework we are calling it through reflection.
+            Type systemEventsType = Type.GetType("Microsoft.Win32.SystemEvents, System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089", throwOnError: false);
+            EventInfo upEventInfo = systemEventsType?.GetEvent("UserPreferenceChanging", BindingFlags.Public | BindingFlags.Static);
+
+            if (upEventInfo != null)
+            {
+                // Delegate TargetType
+                Type userPrefChangingDelegateType = Type.GetType("Microsoft.Win32.UserPreferenceChangingEventHandler, System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089", throwOnError: false);
```

Comment:
If this isn't found, userPrefChangingDelegateType will be null, and then Delegate.CreateDelegate will throw an ArgumentNullException, which will cause everything else to stop working (and be hard to diagnose).  Can we handle that case more gracefully? e.g. just skip the extra initialization if this returns null?  Same for GetMethod below.

Reply:
something like this ?

```C#
if (upEventInfo != null)
{
    // Delegate TargetType
    Type userPrefChangingDelegateType = Type.GetType("Microsoft.Win32.UserPreferenceChangingEventHandler, System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089", throwOnError: false);
    Debug.Assert(userPrefChangingDelegateType != null);

    if (userPrefChangingDelegateType != null)
    {
        // we are using MethodInfo overload because it allows relaxed signature binding i.e. the types dont need to be exact match. It allows base classes as well.
        MethodInfo mi = typeof(KnownColorTable).GetMethod("OnUserPreferenceChanging", BindingFlags.NonPublic | BindingFlags.Static);
        Debug.Assert(mi != null);

        if (mi != null)
        {
            // Creating a delegate to use it as event handler.
            Delegate handler = Delegate.CreateDelegate(userPrefChangingDelegateType, mi);
            upEventInfo.AddEventHandler(null, handler);
        }
    }
}
```